pygit2/settings.py

```diff
@@ -101,4 +101,30 @@ def cache_object_limit(self, object_type, value):
         """
         return option(GIT_OPT_SET_CACHE_OBJECT_LIMIT, object_type, value)
 
+    def __set_ssl_cert_file(self, value):
+        """Set the ssl cert file. The value cannot be read.
+        """
+        return option(GIT_OPT_SET_SSL_CERT_LOCATIONS, value, None)
+
+    ssl_cert_file = property(fset=__set_ssl_cert_file)
```

Comment:
It is an ancient way of creating properties. Always use it as a decorator instead.

Reply:
Also, we should have a getter for this.

Reply:
There is no get support for the ssl cert paths in `libgit2`. It makes sense to have it added to `libgit2` first.

You can not use `property` as decorator if you don't have a setter since the first param in the `__init__` for `property` is `fget`.

Reply:
But you can track it on the Python side because it's needed to properly set the other value.

Reply:
you can still use a decorator:
```python
@property
def attr(self):
    """A proper docstring."""
    raise AttributeError

@attr.setter
def attr(self, value):
    # set val
```

Reply:
cc https://github.com/libgit2/pygit2/pull/879#issuecomment-468791003

So here's why we need to track the values we set earlier. Semantically, these setters are assigned with the values for what their names mean.
End-users of the library are Pythonistas who always expect the assignment behavior of the attribute to change exactly the thing its name refers to.
So resetting another setting would be misleading and is a side-effect.
That's why we need to track values which we set earlier for one connected setting to use them when setting the other.
Of course, it's not necessary to expose them for reading right now. But it's a least we can do to preserve consistent UX when setting one of TSL cert location sub-options.
As for setting both sub-options, it should be an explicit method because semantically setting a tuple to attribute to side-effect two values is weird.

Reply:
Digging around, it looks like ultimately `libgit2` makes a call to `SSL_CTX_load_verify_locations` when built with Openssl and the [man page](https://linux.die.net/man/3/ssl_ctx_load_verify_locations) for it seems to describe the following behavior.

`When looking up CA certificates, the OpenSSL library will first search the certificates in CAfile, then those in CApath.`

Reply:
Nice, that's what I thought...